### DISCUSSION TITLE
Eclipse CDT does not recognize C++20 concepts

### DISCUSSION THREAD

**Discussion by MadRoos** [eclipse-cdt/cdt#762] [Q&A]
Hello,

I am currently using the latest version of Eclipse CDT along with GCC 13.2.0 compiler. I've been trying to utilize C++20 concepts in my code, specifically using the requires keyword, but Eclipse seems to be unable to recognize it. When I include concepts like requires std::totally_ordered<T>, Eclipse expects a semicolon at the end of the line and does not recognize the requires keyword.

Is there a specific configuration or update required to enable support for C++20 concepts in Eclipse CDT? Any guidance on how to resolve this issue would be greatly appreciated.

Thank you.

**Comment by ghentschke** [comment]
Hi, for new C/C++ language features please try the [cdt-lsp](https://github.com/ghentschke/eclipse-cdt-lsp) feature ([clangd](https://clangd.llvm.org/)). It provides a new Language Server (LSP) backed Editor. It can (currently) be used for cmake based projects. Managed Build project support is under development.
